### ISSUE TITLE
`import/order`: treat `type` imports like regular imports?

### ISSUE THREAD

**Issue by alumni** [benmosher/eslint-plugin-import#2070]
How can we treat type imports like a regular import (i.e. the behavior before v2.23.0)? I would expect the following import order to be fine:
```ts
import type { Dictionary } from 'lodash';
import { isEmpty } from 'lodash';

import { something } from './relative';
```
... but the first line gets highlighted:

> There should be no empty line within import group

Also, eslint is now running forever, so I can't automatically fix this.

Rule config:
```json
{
  "import/order": [
    "warn",
    {
      "alphabetize": { "order": "asc" },
      "groups": [
        ["builtin", "external"],
        "internal",
        ["parent", "sibling", "index"],
        "object"
      ],
      "newlines-between": "always",
      "pathGroups": [
        {
          "pattern": "*.json",
          "group": "object",
          "patternOptions": { "matchBase": true }
        }
      ]
    }
  ]
}

```

**Comment by grit96** [comment]
I think the way you'd do that is add `type` to your first group array.

**Comment by alumni** [comment]
> I think the way you'd do that is add `type` to your first group array.

That won't work in this case: 
```ts
import type { Dictionary } from 'lodash';
import { isEmpty } from 'lodash';

import { something } from './relative';
import type { TestType } from './relative';
```

With 2.22.1 the above code will pass, but with 2.23.0 it will fail.

**Comment by ljharb** [comment]
That has "type A, value A, value B, type B", so I'd expect that to fail regardless of how you configure it - ie, it should be "type A, value A, type B, value B", or "type A, type B, value A, value B", or the reverse of either.

**Comment by fernandopasik** [comment]
After 2.23.0 with this example

```
import type { PackageJson } from 'type-fest';
import Generator from 'yeoman-generator';
```

is throwing an error
```
2:1  error  `yeoman-generator` import should occur before import of `type-fest`  import/order
```

Wouldn't it be ok since the alphabetical sorting is right?

**Comment by alumni** [comment]
@ljharb v2.22.1 was not enforcing any order for value and type imports, they were treated identically. v2.23 introduced this breaking change which sorts type imports separately from value imports. Currently, the imports are sorted like this:
```ts
import { isEmpty } from 'lodash';

import { something } from './relative';

import type { TestType } from './relative';
import type { Dictionary } from 'lodash';
```

... which is also not right according to your own expectation.

Since this is a minor version update, it would have been nice to leave the old behavior unchanged. It is a bit late for that now, but it would still be nice to have to ability to sort type and value imports together - either "type A, value A, value B, type B" or "type A, value A, type B, value B".

**Comment by ljharb** [comment]
@alumni yes, I agree with that.

We'll need to restore that default behavior, and only differentiate type imports via an option.

**Comment by merrywhether** [comment]
> That has "type A, value A, value B, type B", so I'd expect that to fail regardless of how you configure it - ie, it should be "type A, value A, type B, value B", or "type A, type B, value A, value B", or the reverse of either.

For the version where types are mixed within groups, it would be nice if ^ was actually enforced. We have a bunch of

```
import type A from 'a';
import A from 'a';
import B from 'b';
import type B from 'b';
```

and it would be great to have a more strict ordering if the library is going to start being type-import aware.

**Comment by mikestopcontinues** [comment]
Ideal for me:

* If `type` is not listed in `groups`, sort them right above the matching imports. (Or below, or whatever.)
* If `type` is listed, sort all type imports at that location AND sort that group in the same way as the overall group structure, sans line-breaks.
* Do not let `import/first` warn for any type imports begin out of order regarding other imports. (Which it's currently doing, if you add `type` to groups.)

This reverts the old behavior for people with old configs, but allows newer configs to leverage the coolness of a `type` group.

**Comment by ahnpnl** [comment]
I got the same issue when upgrading from 2.22.1 to 2.23.2, our CI failed https://github.com/kulshekhar/ts-jest/runs/2598048887?check_suite_focus=true

**Comment by fernandopasik** [comment]
Thanks @ljharb for restoring the behaviour! Is there going to be a release soon?

**Comment by ljharb** [comment]
#2092
